drivers/include/periph/timer.h

```diff
@@ -43,6 +43,13 @@
 extern "C" {
 #endif
 
+/**
```

Comment:
I think a `#include "periph_conf.h"` is missing in this file, otherwise the check for `TIMER_CHANNEL_NUMOF` fails.